readthedocs/api/v2/views/integrations.py

```diff
@@ -441,6 +463,17 @@ def is_payload_valid(self):
             return False
         return token == secret
 
+    def get_external_version_data(self):
+        """Get Commit Sha and merge request number from payload."""
+        try:
+            identifier = self.data['object_attributes']['last_commit']['id']
+            verbose_name = str(self.data['object_attributes']['iid'])
+
+            return identifier, verbose_name
+
+        except KeyError:
+            raise ParseError('Parameters "id" and "iid" are required')
```

Comment:
Where is this `ParseError` handled?

Reply:
When there is a `ParseError` we return `400` to the API. We do the same thing for the `push` events.